src/Sentry/SentrySpan.cs

```diff
@@ -104,6 +105,10 @@ public SentrySpan(ISpan tracer)
         {
             _measurements = spanTracer.InternalMeasurements?.ToDict();
             _tags = spanTracer.InternalTags?.ToDict();
+            if (spanTracer.HasMetrics)
+            {
+                _metricsSummary = spanTracer.MetricsSummary;
+            }
```

Comment:
So the child inherits the summary from the parent?

Reply:
Not quite... I think I've mixed responsibilities here a little. 

I've refactored this now, splitting the LocalAggregator into two separate classes with separate responsibilities: 
1. MetricsSummaryAggregator to track totals
2. MetricsSummary for serialization across the wire

This makes the serialization logic easier to understand as well since the way the metrics are stored either has to be optimized to be easy to mutate or easy to serialize (no easy way to have both, due to the requirement to group by exportKey for serialization).

Reply:
> So the child inherits the summary from the parent?

btw That mirrors the difference between SpanTracer and SentrySpan now as well (the first is mutable, designed to be used when actively tracking a span and the second is an immutable DTO for serialization).

Reply:
Oh nice. The split makes it way easier to wrap my head around it. Thanks!